napari/_qt/widgets/qt_dims_sorter.py

```diff
@@ -46,55 +37,92 @@ def _array_in_range(arr: np.ndarray, low: int, high: int) -> bool:
+
     Modified from:
     https://github.com/jni/zarpaint/blob/main/zarpaint/_dims_chooser.py
+
+    Parameters
+    ----------
+    viewer : napari.Viewer
+        Main napari viewer instance.
+    parent : QWidget
+        QWidget that holds this widget. A QtDimsSorter instances will
+        disconnect all callbacks upon closing of it's parent.
+
+    Attributes
+    ----------
+    axis_list : napari._qt.containers.qt_axis_model.AxisList
+        Selectable evented list representing the viewer axes.
     """
 
-    def __init__(self, viewer: 'Viewer', parent=None) -> None:
+    def __init__(self, viewer: 'Viewer', parent: QWidget) -> None:
         super().__init__(parent=parent)
         dims = viewer.dims
-        root = SelectableEventedList(
-            [AxisModel(dims, dims.order[i]) for i in range(dims.ndim)]
+        self.axis_list = AxisList.from_dims(dims)
+
+        view = QtListView(self.axis_list)
+        view.setSizeAdjustPolicy(QtListView.AdjustToContents)
+        locked_icon_path = f'{USER_CACHE_DIR}/_themes/{viewer.theme}/lock.svg'
+        unlocked_icon_path = (
+            f'{USER_CACHE_DIR}/_themes/{viewer.theme}/lock_open.svg'
         )
-        root.events.reordered.connect(
-            lambda event: set_dims_order(dims, event.value)
+        view.setStyleSheet(
+            "QListView::indicator:unchecked"
+            "{"
+            f"image: url({locked_icon_path});"
+            "}"
+            "QListView::indicator:checked"
+            "{"
+            f"image: url({unlocked_icon_path});"
+            "}"
         )
-        dims.events.order.connect(
-            lambda event: move_indices(root, event.value)
-        )
-        view = QtListView(root)
-        view.setSizeAdjustPolicy(QtListView.AdjustToContents)
-
-        self.axes_list = root
 
         layout = QGridLayout()
         self.setLayout(layout)
 
         widget_tooltip = QtToolTipLabel(self)
         widget_tooltip.setObjectName('help_label')
-        widget_tooltip.setToolTip(trans._('Drag dimensions to reorder.'))
+        widget_tooltip.setToolTip(
+            trans._(
+                'Drag dimensions to reorder, click lock icon to lock dimension in place.'
+            )
+        )
 
         widget_title = QLabel(trans._('Dims. Ordering'), self)
 
         self.layout().addWidget(widget_title, 0, 0)
         self.layout().addWidget(widget_tooltip, 0, 1)
-        self.layout().addWidget(view, 1, 0)
+        self.layout().addWidget(view, 1, 0, 1, 2)
+
+        # connect axis_list and dims
+        # terminate connections after parent widget is closed
+        # to allow closure of QtDimsSorter
+        self.axis_list.events.reordered.connect(
+            lambda event: set_dims_order(dims, event.value),
+            until=self.parent().destroyed,
+        )
+        dims.events.order.connect(
+            lambda event: move_indices(self.axis_list, event.value),
+            until=self.parent().destroyed,
+        )
```

Comment:
```suggestion
        # connect axis_list and dims
        # terminate connections after parent widget is closed
        # to allow closure of QtDimsSorter
        self.axis_list.events.reordered.connect(
            self.set_dims_order
        )
        dims.events.order.connect(
            self.move_indices
        )
        
    def set_dims_order(self, event):
        set_dims_order(self.dims, event.value)
    
    def move_indices(self, event):
        move_indices(self.axis_list, event.value)
```

It is enough to use methods to get this behavior, as it will be deleted on object clean.